holoviews/element/graphs.py

```diff
@@ -392,3 +392,119 @@ class EdgePaths(Path):
+            try:
+                # Try assuming data contains indices (3 columns)
+                nodes = self._node_type(nodes)
+            except:
+                # Try assuming data contains just coordinates (2 columns)
+                try:
+                    points = Points(nodes)
+                    ds = Dataset(points).add_dimension('index', 2, np.arange(len(points)))
+                    nodes = self._node_type(ds)
+                except:
+                    raise ValueError("Nodes argument could not be interpreted, expected "
+                                     "data with two or three columns representing the "
+                                     "x/y positions and optionally the node indices.")
+        if edgepaths is not None and not isinstance(edgepaths, self._edge_type):
+            edgepaths = self._edge_type(edgepaths)
+        super(TriMesh, self).__init__(edges, kdims=kdims, vdims=vdims, **params)
+        self._nodes = nodes
+        self._edgepaths = edgepaths
+
+    @classmethod
+    def from_points(cls, points):
+        """
+        Uses Delauney triangulation to compute triangle simplices for
+        each point.
+        """
+        try:
+            from scipy.spatial import Delaunay
+        except:
+            raise ImportError("Generating triangles from points requires, "
+                              "SciPy to be installed.")
+        if not isinstance(points, Points):
+            points = Points(points)
+        tris = Delaunay(points.array([0, 1]))
+        return cls((tris.simplices, points))
+
+    @property
+    def edgepaths(self):
+        """
+        Returns the EdgePaths by generating a triangle for each simplex.
+        """
+        if self._edgepaths:
+            return self._edgepaths
+        elif not len(self):
+            edgepaths = self._edge_type([], kdims=self.nodes.kdims[:2])
+            self._edgepaths = edgepaths
+            return edgepaths
+
+        simplices = self.array([0, 1, 2]).astype(np.int32)
+        pts = self.nodes.array([0, 1]).astype(float)
+        empty = np.array([[np.NaN, np.NaN]])
+        paths = [arr for tri in pts[simplices] for arr in
+                 (tri[[0, 1, 2, 0], :], empty)][:-1]
+        edgepaths = self._edge_type([np.concatenate(paths)],
+                                    kdims=self.nodes.kdims[:2])
+        self._edgepaths = edgepaths
+        return edgepaths
+
+    def select(self, selection_specs=None, **selection):
+        """
+        Allows selecting data by the slices, sets and scalar values
+        along a particular dimension. The indices should be supplied as
+        keywords mapping between the selected dimension and
+        value. Additionally selection_specs (taking the form of a list
+        of type.group.label strings, types or functions) may be
+        supplied, which will ensure the selection is only applied if the
+        specs match the selected object.
+        """
+        self.edgepaths
```

Comment:
Will add a comment about this. By default ``edgepaths`` are not computed but .select expects it to be.

Reply:
Yes, a comment next to this line explaining it would be a good idea.